src/DDTrace/Integrations/ElasticSearch/V1/ElasticSearchIntegration.php

```diff
@@ -0,0 +1,328 @@
+            $span = $scope->getSpan();
+
+            $span->setTag(Tags\SERVICE_NAME, ElasticSearchIntegration::DEFAULT_SERVICE_NAME);
+            $span->setTag(Tags\SPAN_TYPE, Types\ELASTICSEARCH);
+            $span->setTag(Tags\RESOURCE_NAME, ElasticSearchIntegration::buildResourceName($name, $params));
+
+            // PHP 5.4 compatible try-catch-finally
+            $thrown = null;
+            $result = null;
+            try {
+                $result = call_user_func_array([$this, $name], $args);
+            } catch (\Exception $ex) {
+                $thrown = $ex;
+                if ($span instanceof Span) {
+                    $span->setError($ex);
+                }
+            }
+            $scope->close();
+            if ($thrown) {
+                throw $thrown;
+            }
+
+            return $result;
+        });
+    }
+
+    /**
+     * @param string $class
+     * @param array $methods
+     */
+    public static function traceSimpleMethodsCall($class, array $methods)
+    {
+        foreach ($methods as $method) {
+            ElasticSearchIntegration::traceMethod($class, $method);
+        }
+    }
+
+    /**
+     * @param string $class
+     * @param string $name
+     */
+    public static function traceMethod($class, $name)
+    {
+        if (!method_exists($class, $name)) {
+            return;
+        }
+
+        dd_trace($class, $name, function () use ($class, $name) {
+            $args = func_get_args();
+
+            $tracer = GlobalTracer::get();
+            $operationName = str_replace('\\', '.', "$class.$name");
+            $scope = $tracer->startActiveSpan($operationName);
+            $span = $scope->getSpan();
+
+            $span->setTag(Tags\SERVICE_NAME, ElasticSearchIntegration::DEFAULT_SERVICE_NAME);
+            $span->setTag(Tags\SPAN_TYPE, Types\ELASTICSEARCH);
+            $span->setTag(Tags\RESOURCE_NAME, $operationName);
+
+            // PHP 5.4 compatible try-catch-finally
+            $thrown = null;
+            $result = null;
+            try {
+                $result = call_user_func_array([$this, $name], $args);
+            } catch (\Exception $ex) {
+                $thrown = $ex;
+                if ($span instanceof Span) {
+                    $span->setError($ex);
+                }
+            }
+            $scope->close();
+            if ($thrown) {
+                throw $thrown;
+            }
+
+            return $result;
+        });
+    }
+
+    /**
+     * @param string $namespace
+     * @param string $name
+     */
+    public static function traceNamespaceMethod($namespace, $name)
+    {
+        $class = 'Elasticsearch\Namespaces\\' . $namespace;
+        if (!method_exists($class, $name)) {
+            return;
+        }
+
+        dd_trace($class, $name, function () use ($namespace, $name) {
+            $args = func_get_args();
+            list($params) = extract($args);
```

Comment:
Same thing here and replacing `$params` with `$args` on line 283. :)

Reply:
Same as above